pkg/csi_driver/multishare_controller.go

```diff
@@ -86,6 +87,34 @@ func (m *MultishareController) CreateVolume(ctx context.Context, req *csi.Create
 		return nil, err
 	}
 
+	// If we are creating a new instance, we need pick an unused CIDR range from reserved-ipv4-cidr
+	// If the param was not provided, we default reservedIPRange to "" and cloud provider takes care of the allocation
+	param := req.GetParameters()
+	if instance.Network.ConnectMode == privateServiceAccess {
+		if reservedIPRange, ok := param[paramReservedIPRange]; ok {
+			instance.Network.ReservedIpRange = reservedIPRange
+		}
+	} else if reservedIPV4CIDR, ok := param[paramReservedIPV4CIDR]; ok {
+		reservedIPRange, err := m.server.reserveIPRange(ctx, &file.ServiceInstance{
```

Comment:
this will not work correctly if multishare is using a non-prod env. reserverIpRange -> getCloudInstancesReservedIPRanges -> s.config.fileService.ListInstances(ctx, filer). 
(ListInstances() works only for prod env)

Let's discuss this a bit more

Reply:
conversely I also see that ListInstances() -> cloudInstanceToServiceInstance() which reads instance.FileShares[0].Name.  This field will not be valid for multishare instances.

reserveip should handle both types of instances (single and multishare instances since they all fall under enterprise tier). ListMultishareInstances()->cloudInstanceToMultishareInstance(), ListInstances() -> cloudInstanceToServiceInstance()

Reply:
Removed commit from pr since it needs more investigation